### PR TITLE
CMS Kit - Tag and media refactorings

### LINKED ISSUE #7835: CMS Kit - Tag and media refactorings
See the tags and sections in the meeting notes.

### PR DESCRIPTION
**Pull request by enisn**
- Closes  #7835

### CODE REVIEW

File: modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Tags/EntityTagAdminAppService.cs [hunk 1]
```diff
@@ -57,24 +57,11 @@ public virtual async Task RemoveTagFromEntityAsync(EntityTagRemoveDto input)
...
-
-            await EntityTagRepository.DeleteManyAsync(deletedTags.Select(s => s.Id).ToArray());
-
-            foreach (var addedTag in addedTags)
-            {
-                var tag = await TagManager.GetOrAddAsync(input.EntityType, addedTag, CurrentTenant?.Id);
-
-                await EntityTagManager.AddTagToEntityAsync(tag.Id, input.EntityType, input.EntityId, CurrentTenant?.Id);
-            }
+            await this.EntityTagManager.SetEntityTagsAsync(input.EntityType, input.EntityId, input.Tags);
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Tags/EntityTagAdminAppService.cs]: No need to use `this`

File: modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Tags/TagAdminAppService.cs [hunk 2]
```diff
@@ -47,8 +49,7 @@ public override async Task<TagDto> CreateAsync(TagCreateDto input)
             var tag = await TagManager.InsertAsync(
                 GuidGenerator.Create(),
                 input.EntityType,
-                input.Name,
-                CurrentTenant?.Id);
+                input.Name);
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Tags/TagAdminAppService.cs:52]: It should be good to create the entity-tag first then insert it.

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs [hunk 3]
```diff
@@ -18,56 +19,46 @@ public TagManager(ITagRepository tagRepository, ITagDefinitionStore tagDefinitio
...
-            return entity;
+            return await TagRepository.FindAsync(entityType, name)
+                    ?? await InsertAsync(GuidGenerator.Create(), entityType, name);
         }
 
-        public virtual async Task<Tag> InsertAsync(Guid id, [NotNull] string entityType, [NotNull] string name,
-            Guid? tenantId = null, CancellationToken cancellationToken = default)
+        public virtual async Task<Tag> InsertAsync(Guid id,
+                                                   [NotNull] string entityType,
+                                                   [NotNull] string name)
```
Comment (cotur) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs]: Those can stay on one line.

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs [hunk 4]
```diff
@@ -18,56 +19,46 @@ public TagManager(ITagRepository tagRepository, ITagDefinitionStore tagDefinitio
...
-            if (await TagRepository.AnyAsync(entityType, name, tenantId, cancellationToken))
+            if (await TagRepository.AnyAsync(entityType, name))
             {
                 throw new TagAlreadyExistException(entityType, name);
             }
 
-            return await TagRepository.InsertAsync(new Tag(id, entityType, name, tenantId),
-                cancellationToken: cancellationToken);
+            return await TagRepository.InsertAsync(
+                new Tag(id, entityType, name, CurrentTenant.Id));
```
Comment (cotur) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs]: not necessary newline

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs [hunk 5]
```diff
@@ -18,56 +19,46 @@ public TagManager(ITagRepository tagRepository, ITagDefinitionStore tagDefinitio
...
         }
 
-        public virtual async Task<Tag> UpdateAsync(Guid id, [NotNull] string name,
-            CancellationToken cancellationToken = default)
+        public virtual async Task<Tag> UpdateAsync(Guid id,
+                                                   [NotNull] string name)
```
Comment (cotur) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs]: not necessary newline

File: modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Tags/EntityTagAdminAppService.cs (hunk 1 above)
Comment (enisn) [inline modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Tags/EntityTagAdminAppService.cs]:
```suggestion
            await EntityTagManager.SetEntityTagsAsync(input.EntityType, input.EntityId, input.Tags);
```

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs (hunk 3 above)
Comment (enisn) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs]:
```suggestion
        public virtual async Task<Tag> InsertAsync(Guid id, [NotNull] string entityType, [NotNull] string name)
```

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs [hunk 6]
```diff
@@ -18,56 +19,53 @@ public TagManager(ITagRepository tagRepository, ITagDefinitionStore tagDefinitio
...
-            return await TagRepository.InsertAsync(new Tag(id, entityType, name, tenantId),
-                cancellationToken: cancellationToken);
+            return
+                new Tag(id, entityType, name, CurrentTenant.Id);
         }
 
-        public virtual async Task<Tag> UpdateAsync(Guid id, [NotNull] string name,
-            CancellationToken cancellationToken = default)
+        public virtual async Task<Tag> UpdateAsync(Guid id,
+                                                   [NotNull] string name)
```
Comment (enisn) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs]:
```suggestion
        public virtual async Task<Tag> UpdateAsync(Guid id, [NotNull] string name)
```

Comment (enisn) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs]:
```suggestion
        public virtual async Task<Tag> UpdateAsync(Guid id, [NotNull] string name)
```

Comment (enisn) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/TagManager.cs]:
```suggestion
        public virtual async Task<Tag> UpdateAsync(Guid id, [NotNull] string name)
```

**Review by ilkayilknur — APPROVED** [review]

**Pull request merged**